text/0000-hyphenation-data-api.md

```diff
@@ -0,0 +1,111 @@
+# RFC Template
+
+- Start Date: 2025-02-20
+- RFC PR: [electron/rfcs#0000](https://github.com/electron/rfcs/pull/0000)
+- Electron Issues: https://github.com/electron/electron/issues/33692
+- Reference Implementation: 
+- Status: **Draft**
+
+# Hyphenation data support
+
+## Summary
+
+Chromium uses a folder called `hyphen-data` downloaded after installation to enable automatic hyphenation of words in various 
+languages. This RFC is to make the same hyphenation feature available to Electron apps.
+
+## Motivation
+
+One of the promises of Electron is that it renders the same as Chromium. Without hyphenation, this is not the case for 
+languages like German or Dutch. As a result, text in these languages is not hyphenated across different text lines, 
+causing the text to have different dimensions, break at different places and generally look worse than in Chromium.
+
+Adding support for hyphenation data will bring the rendering of long words across text lines in line with the rendering 
+in Chromium
+
+## Guide-level explanation
+
+We propose an API for hyphenation data similar to the existing [spellChecker API](https://www.electronjs.org/docs/latest/tutorial/spellchecker) in Electron. The reason for this is that spell checking data in Chromium is similarly downloaded on launch (rather than shipping with the renderer) and Electron replicates this behavior.
+
+### Enabling hyphenation
+
+To enable hyphenation in your Electron app, you can use the following code:
+
+```javascript
+const myWindow = new BrowserWindow({
+  webPreferences: {
+    hyphenation: true
```

Comment:
For spellchecker there is also a [ses.setSpellCheckerEnabled](https://www.electronjs.org/docs/latest/api/session#sessetspellcheckerenabledenable) function which allows the spellchecker to be enabled at the session level. Would it be helpful to also expose an API for enabling/disabling hyphenation at the session level?

Reply:
I guess so, though hyphenation is a lot more "static" than spellchecking (which you might want to change during the lifetime of an app). I'll add the associated methods (and events) to the RFC regardless.